Anime-App/Feature/Anime/View/AnimeViewController.swift

```diff
@@ -129,6 +138,42 @@ extension AnimeViewController {
         
     }
     
+    func fetchAnime() {
+        self.spinner.startAnimating()
+        
+        animeViewModel.fetchAnime(page: page) { [weak self] res in
+            switch res {
+            case .success:
+                self?.tableView.reloadData()
+                self?.isLoading = false
+            case .failure:
+                print("no more data")
+             }
+            self?.spinner.stopAnimating()
+        }
+    }
+    
+}
+
+
+extension AnimeViewController: UITableViewDataSourcePrefetching {
+
+    func tableView(_ tableView: UITableView, prefetchRowsAt indexPaths: [IndexPath]) {
+        for indexPath in indexPaths {
+            let lastItem = animeViewModel.animes.count - 1
+            if indexPath.row == lastItem {
+                page+=2
```

Comment:
Why are you adding 2 to the page? You're going to be doing 2, 4, 6, 8. When you read a book do you not read it page 1, 2, 3, 4 or 2, 4, 6, 8.